csharp/src/Ice/Internal/WSEndpoint.cs

```diff
@@ -34,40 +34,7 @@ internal WSEndpoint(ProtocolInstance instance, EndpointI del, Ice.InputStream s)
         _resource = s.readString();
     }
 
-    private sealed class InfoI : Ice.WSEndpointInfo
-    {
-        public InfoI(EndpointI e)
-        {
-            _endpoint = e;
-        }
-
-        public override short type()
-        {
-            return _endpoint.type();
-        }
-
-        public override bool datagram()
-        {
-            return _endpoint.datagram();
-        }
-
-        public override bool secure()
-        {
-            return _endpoint.secure();
-        }
-
-        private EndpointI _endpoint;
-    }
-
-    public override Ice.EndpointInfo getInfo()
-    {
-        Ice.WSEndpointInfo info = new InfoI(this);
-        info.underlying = _delegate.getInfo();
-        info.resource = _resource;
-        info.compress = info.underlying.compress;
-        info.timeout = info.underlying.timeout;
-        return info;
-    }
+    public override EndpointInfo getInfo() => new WSEndpointInfo(_delegate.getInfo(), _resource);
```

Comment:
It's not clear to me how the secure, type, datagram, timeout, compress fields of this returned `WSEndpointInfo` are set. 

They should be set to the underlying endpoint info values.

Reply:
See the base class, Ice.EndpointInfo.

type(), datagram() and secure() delegate to underlying, while the timeout and compress fields are set in the constructor from underlying.